pytools/py_codegen.py

```diff
@@ -35,42 +41,68 @@
 
 
 class PythonCodeGenerator(CodeGeneratorBase):
-    def get_module(self, name=None):
+    def get_module(self, name: str | None = None,
+                   _from_get_function: bool = False) -> dict[str, Any]:
         if name is None:
             name = "<generated code>"
 
-        result_dict = {}
+        result_dict: dict[str, Any] = {}
         source_text = self.get()
+
+        # {{{ Handle Python's linecache
+
+        import linecache
+
+        if name in linecache.cache:
+            from warnings import warn
+            warn(f"Overwriting existing generated code in linecache: '{name}'.",
+                    stacklevel=3 if _from_get_function else 2)
```

Comment:
Since you catch these warnings below in the tests: could you introduce a specific type for them? (And use that with `pytest.warns`)

Reply:
What do you think of 8874bc0?